File: api/src/main/java/net/asch/bulkit/api/network/GuiPayloads.java
package net.asch.bulkit.api.network;

import net.asch.bulkit.api.BulkIt;
import net.minecraft.network.RegistryFriendlyByteBuf;
import net.minecraft.network.codec.StreamCodec;
import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
import org.jetbrains.annotations.NotNull;

public class GuiPayloads {
    public record Disk() implements CustomPacketPayload {
        public static final Disk INSTANCE = new Disk();
        public static final Type<Disk> TYPE = new Type<>(BulkIt.location("disk_gui"));
        public static final StreamCodec<RegistryFriendlyByteBuf, Disk> STREAM_CODEC = StreamCodec.unit(INSTANCE);

        @Override
        public @NotNull Type<? extends CustomPacketPayload> type() {
            return TYPE;
        }
    }
}

File: api/src/main/kotlin/net/asch/bulkit/api/network/GuiPayloads.kt
package net.asch.bulkit.api.network

import net.asch.bulkit.api.BulkItApi
import net.minecraft.network.RegistryFriendlyByteBuf
import net.minecraft.network.codec.StreamCodec
import net.minecraft.network.protocol.common.custom.CustomPacketPayload

object GuiPayloads {
    class Disk : CustomPacketPayload {
        override fun type(): CustomPacketPayload.Type<out CustomPacketPayload?> {
            return TYPE
        }

        companion object {
            val INSTANCE: Disk = Disk()
            val TYPE: CustomPacketPayload.Type<Disk> = CustomPacketPayload.Type(BulkItApi.location("disk_gui"))
            val STREAM_CODEC: StreamCodec<RegistryFriendlyByteBuf, Disk> = StreamCodec.unit(INSTANCE)
        }
    }
}